torch/_dynamo/variables/distributed.py

```diff
@@ -148,9 +148,36 @@ def is_device_mesh(value):
     def as_python_constant(self):
         return self.value
 
+    def call_method(
+        self,
+        tx,
+        name,
+        args: "List[VariableTracker]",
+        kwargs: "Dict[str, VariableTracker]",
+    ) -> "VariableTracker":
+        if name == "get_dim_groups":
+            args = [x.as_python_constant() for x in args]
+            kwargs = {k: v.as_python_constant() for k, v in kwargs.items()}
+            dim_groups = self.value.get_dim_groups(*args, **kwargs)
+            # desugar the results to ProcessGroupVariables
+            options = VariableTracker.propagate(self)
```

Comment:
where do you propagate guards from the args and kwargs?

Reply:
Good point, I'll update